react/logic/masterDataError.ts

```diff
@@ -0,0 +1,112 @@
+import { ApolloError } from 'apollo-client'
+import { GraphQLError } from 'graphql'
+import { ErrorTypes, concatFormPath } from 'react-hook-form-jsonschema'
+
+export type MasterDataErrorRecord = Record<string, ErrorTypes[]>
+
+interface MasterDataError {
+  Message: string
+  LineNumber: number
+  LinePosition: number
+  Path: string
+  Value: any
+  SchemaId: string
+  SchemaBaseUri: string | null
+  ErrorType: string
+  ChildErrors: []
+}
+
+const concatErrors = (
+  errors: MasterDataError[],
+  error: { schema: string; errors: MasterDataError }
+) => {
+  if (error.errors) {
+    return errors.concat(error.errors)
+  }
+  return errors
+}
+
+const filterMasterDataErrors = (
+  acc: MasterDataError[],
+  message: GraphQLError
+) => {
+  if (message?.extensions?.exception?.response?.data?.errors) {
+    return acc.concat(
+      message.extensions.exception.response.data.errors.reduce(concatErrors, [])
+    )
+  }
+  return acc
+}
+
+const createOrPushError = (
+  errorRecord: MasterDataErrorRecord,
+  path: string,
+  error: ErrorTypes
+) => {
+  if (errorRecord[path]) {
+    errorRecord[path].push(error)
+  } else {
+    errorRecord[path] = [error]
+  }
+}
+
+const evaluateMasterDataRequiredErrors = (
+  data: { nodes: MasterDataErrorRecord; schemaId: string },
+  nodeName: string
+) => {
+  const path = concatFormPath(
+    concatFormPath(data.schemaId, 'properties'),
+    nodeName
+  )
+  createOrPushError(data.nodes, path, ErrorTypes.required)
+  return { nodes: data.nodes, schemaId: data.schemaId }
+}
+
+const evaluateMasterDataErrors = (
+  acc: MasterDataErrorRecord,
+  serverError: MasterDataError
+) => {
+  switch (serverError.ErrorType) {
+    case 'required':
+      acc = (serverError.Value as Array<string>).reduce(
```

Comment:
I see that `Value` has type `any` is it really any or is always an `Array` of something?

Reply:
So, as there is no documentation of it, based on trial and error it returns whatever data it was sent and raised an error, except in the case of required where it always returns an array of strings.

Reply:
In this case I think is better to verify if `Value` is an array before trying to call reduce